### PR TITLE
WV-2638: Dynamic Data Visualization -updated

### PR DESCRIPTION
**Pull request by ryanweiler92**
## Description

This is the implementation of the UI and further functionality for the dynamic data visualization layers that we are utilizing through the FIRMS API. This adds some layers with preset RGB band combinations and also a landsat customizable layer. This customizable layer allows for users to select custom RGB combinations. 

This imagery follows orbit tracks that can be added through the associated layers settings. In the instructions below I've added a URL with a good starting point. But if you want to find imagery on other dates, simply locate the orbit tracks on the date of interest (United States Only). 

I have also built out some tool components that can be used by developers to help debug and test features as they are built. This mode can only be engaged by setting the `tileImageTestMode` state variable in the `mapUI.js` file to true. 

## To Do

- [x] Add zot for under zoom levels
- [x] Add associated layer name in sidebar
- [x] Complete sentinel presets
- [x] Dynamic Wavelength/Bandwidth/Resolution Attributes
- [x] Update layer names (in [this PR](https://github.com/nasa-gibs/worldview/pull/4522))
- [x] Add layer previews (in [this PR](https://github.com/nasa-gibs/worldview/pull/4522))
- [x] Disable Snapshots for DDV Layers
- [x] Fix overzoom indicator zot
- [x] Fix naming conventions
- [x] Fix HLS_Natural_Color_Landsat showing up as Sentinel 2 instead of Landsat
- [x] Add Sentinel layer for 6-5-4 Vegetative Analysis False Color?
- [x] Limit Landsat Band Options to 1-9

## How To Test

1. `git checkout dynamic-data-viz-exp`
2. `npm ci`
3. `npm run build`
4. `npm run watch`
5. For simplicity, use this [URL](http://localhost:3000/?v=-77.27333274057173,38.77605197284668,-76.83722700850396,38.97116386026658&l=Reference_Labels_15m,Reference_Features_15m,MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&t=2023-04-17-T12%3A21%3A36Z) as a starting point. This will place you over Washington DC on April 17th where any of the new DDV layers will be visible. 
6. Go into the layer picker and add and remove each preset DDV layer (HLS Natural Color, HLS Color Infrared, HLS False Color Vegetation, HLS False Color Urban, HLS Shortwave Infrared). For each layer, wait for the tiles to load and add the associated orbit track in the layer options. 
7. Add the 'HLS Customizable Landsat' layer. 
8. Open the layer options and verify that the current band selection defaults to RED: B07 GREEN: B05 BLUE: B04
9. Click on the customize button within the layer settings and confirm the new customization menu opens. 
10. Create a few custom combinations by changing the values of R, G & B in the bottom selection and then hitting the 'Confirm Band Selection' button. NOTE: Some combinations do not exist! Especially when using the B8A band and higher. We will have to figure out a way to tell what bands can actually be used in any combination. This was _not_ a problem before adding bands above B08.
11. This should close the customization menu and you should shortly be able to visualize your changes on the map. 
12. After testing a few custom combinations, try selecting and confirming the preset values from the customizable menu. 
13. Verify that the preset preview images look similar to the visualization you see on the map.

### CODE REVIEW

File: web/js/components/layer/settings/layer-settings.js [hunk 1]
```diff
@@ -340,6 +341,7 @@ class LayerSettings extends React.Component {
     } = this.props;
     const hasAssociatedLayers = layer.associatedLayers && layer.associatedLayers.length;
     const hasTracks = layer.orbitTracks && layer.orbitTracks.length;
+    const ttilerLayer = layer.id === 'HLS_Customizable_Sentinel' || layer.id === 'HLS_Customizable_Landsat';
```
Comment (PatchesMaps) [inline web/js/components/layer/settings/layer-settings.js:344]: ttileLayer?

**Review by minniewong — APPROVED** [review]
This is so cool!! I'll work with you on the layer names, preview images, layer descriptions, resolution/wavelength/bandwidth. Will we also add Sentinel 2A & B presets in the layer picker like you have for Landsat?

Awesome [burned area/active burning fire front](http://localhost:3000/?v=-76.73829949083444,52.625284051308164,-75.37942904269707,53.23271197460986&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m(hidden),HLS_Customizable_Sentinel(bands=B12;B8A;B04),HLS_S30_Nadir_BRDF_Adjusted_Reflectance,VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor(hidden),MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor(hidden)&lg=true&t=2023-06-25-T19%3A11%3A58Z) image for kicks: 
Screenshot 2023-06-28 at 4 03 49 PM

**Comment by Tomcariello** [comment]
IDK if this is out of scope, but I encountered some errors with comparison mode. From the link you provided, add the customizable layer, start comparison mode & change the date on either the A or B side to trigger the error.

**Comment by ryanweiler92** [comment]
> IDK if this is out of scope, but I encountered some errors with comparison mode. From the link you provided, add the customizable layer, start comparison mode & change the date on either the A or B side to trigger the error.

This should be fixed with the latest commit.

**Review by Tomcariello — APPROVED** [review]
Sorry for the delay, looks great!

**Comment by minniewong** [comment]
@ryanweiler92 It seems HLS_Natural_Color_Landsat is actually showing Sentinel 2 and not Landsat. Could you check the configuration of that? It is showing the [same footprint as Sentinel 2](http://localhost:3000/?v=-123.06158876768633,36.93524996797248,-119.2415575102494,38.81108187179218&l=Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m,HLS_Natural_Color_Sentinel(bands=B04;B03;B02),HLS_L30_Nadir_BRDF_Adjusted_Reflectance(hidden),HLS_S30_Nadir_BRDF_Adjusted_Reflectance(hidden),HLS_Natural_Color_Landsat(bands=B04;B03;B02),VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor(hidden),MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor(hidden)&lg=true&t=2023-07-18-T18%3A17%3A02Z)

Would it be possible to also add a layer that is Sentinel 2A &B, Bands 6-5-4 (Vegetative Analysis, False Color)? This will match the existing one for Landsat. I tried setting up a json file for that, but it doesn't seem to work. (you can check [my branch](https://github.com/nasa-gibs/worldview/pull/4522) if you want to see how I've set it up)

**Comment by minniewong** [comment]
The overzoom indicator zot is incorrect for the DDV layers - they should be the same as the original HLS layers, but currently appear to say the DDV layer is 6x overzoomed when the original HLS layer is at the optimum zoom level.

**Comment by minniewong** [comment]
I guess we should disable image snapshots for the DDV layers

**Review by minniewong — APPROVED** [review]
Everything seems to be working well!

**Pull request merged**